plutus-core/untyped-plutus-core/test/Evaluation/Builtins/Definition.hs

```diff
@@ -319,6 +323,234 @@ test_IdBuiltinData =
+
+    -- ChooseData
+    let actualExp = mkIterApp ()
+                      (tyInst () (apply () caseData $ cons $ I 3) bool)
+                      [ -- constr
+                        runQuote $ do
+                              a1 <- freshName "a1"
+                              a2 <- freshName "a2"
+                              pure $ lamAbs () a1 integer $ lamAbs () a2 (TyApp () Builtin.list dataTy) false
+                        -- map
+                      , runQuote $ do
+                              a1 <- freshName "a1"
+                              pure $ lamAbs () a1 (TyApp () Builtin.list $ mkIterTyApp () pair [dataTy,dataTy]) false
+                       -- list
+                      , runQuote $ do
+                              a1 <- freshName "a1"
+                              pure $ lamAbs () a1 (TyApp () Builtin.list dataTy) false
+                       -- I
+                      , runQuote $ do
+                              a1 <- freshName "a1"
+                              pure $ lamAbs () a1 integer true
+                        -- B
+                      , runQuote $ do
+                              a1 <- freshName "a1"
+                              pure $ lamAbs () a1 (mkTyBuiltin @_ @ByteString ()) false
+                      ]
+
+    Right (EvaluationSuccess true)  @=?  typecheckEvaluateCekNoEmit defaultCekParameters actualExp
+
+-- | Test all cryptography-related builtins
+test_Crypto :: TestTree
+test_Crypto = testCase "Crypto" $ do
+    evals True VerifySignature
+        [ -- pubkey
+          cons @ByteString "Y\218\215\204>\STX\233\152\251\243\158'm\130\&0\197\DEL\STXd\214`\147\243y(\234\167=kTj\164"
+          -- message
+        , cons @ByteString "hello world"
+          -- signature
+        , cons @ByteString "\a'\198\r\226\SYN;\bX\254\228\129n\131\177\193\DC3-k\249RriY\221wIL\240\144\r\145\195\191\196]\227\169U(\ETX\171\SI\199\163\138\160\128R\DC4\246n\142[g\SI\169\SUB\178\245\166\&0\243\b"
+        ]
+
+    evals False VerifySignature
+        [ -- pubkey
+          cons @ByteString "Y\218\215\204>\STX\233\152\251\243\158'm\130\&0\197\DEL\STXd\214`\147\243y(\234\167=kTj\164"
+          -- message
+        , cons @ByteString "HELLO WORLD"
+          -- signature
+        , cons @ByteString "\a'\198\r\226\SYN;\bX\254\228\129n\131\177\193\DC3-k\249RriY\221wIL\240\144\r\145\195\191\196]\227\169U(\ETX\171\SI\199\163\138\160\128R\DC4\246n\142[g\SI\169\SUB\178\245\166\&0\243\b"
+        ]
+    evals @ByteString "\185M'\185\147M>\b\165.R\215\218}\171\250\196\132\239\227zS\128\238\144\136\247\172\226\239\205\233"
+        Sha2_256 [cons @ByteString "hello world"]
+    evals @ByteString "dK\204~VCs\EOT\t\153\170\200\158v\"\243\202q\251\161\217r\253\148\163\FS;\251\242N98"
+        Sha3_256 [cons @ByteString "hello world"]
+    evals @ByteString "%l\131\178\151\DC1M \ESC0\ETB\159?\SO\240\202\206\151\131b-\165\151C&\180\&6\ETB\138\238\246\DLE"
+        Blake2b_256 [cons @ByteString "hello world"]
```

Comment:
Did you check that these outputs are actually what we expect?

Reply:
Added comments:

```
    -- independently verified by `/usr/bin/sha256sum` with the hex output converted to ascii text
    -- sha256sum hex output: b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
    evals @ByteString "\185M'\185\147M>\b\165.R\215\218}\171\250\196\132\239\227zS\128\238\144\136\247\172\226\239\205\233"
        Sha2_256 [cons @ByteString "hello world"]
    -- independently verified by `/usr/bin/sha3-256sum` with the hex output converted to ascii text
    -- sha3-256sum hex output: 644bcc7e564373040999aac89e7622f3ca71fba1d972fd94a31c3bfbf24e3938
    evals @ByteString "dK\204~VCs\EOT\t\153\170\200\158v\"\243\202q\251\161\217r\253\148\163\FS;\251\242N98"
        Sha3_256 [cons @ByteString "hello world"]
    -- independently verified by `/usr/bin/b2sum -l 256` with the hex output converted to ascii text
    -- b2sum -l 256 hex output: 256c83b297114d201b30179f3f0ef0cace9783622da5974326b436178aeef610
    evals @ByteString "%l\131\178\151\DC1M \ESC0\ETB\159?\SO\240\202\206\151\131b-\165\151C&\180\&6\ETB\138\238\246\DLE"
        Blake2b_256 [cons @ByteString "hello world"]
```